cmd/ingester/app/consumer/consumer.go

```diff
@@ -146,6 +147,7 @@ func (c *Consumer) handleMessages(pc sc.PartitionConsumer) {
 
 			if msgProcessor == nil {
 				msgProcessor = c.processorFactory.new(pc.Partition(), msg.Offset-1)
+				msgErrors = msgProcessor.Errors()
```

Comment:
this feels dirty - you're already selecting from msgErrors in L160, but it's not yet initialized